telegram-bot-bumblebee-core/src/main/java/com/github/bumblebee/command/youtube/YoutubeSubscribeCommand.java

```diff
@@ -0,0 +1,107 @@
+package com.github.bumblebee.command.youtube;
+
+import com.github.bumblebee.command.SingleArgumentCommand;
+import com.github.bumblebee.command.youtube.api.YoutubeSubscriptionProvider;
+import com.github.bumblebee.command.youtube.entity.Chat;
+import com.github.bumblebee.command.youtube.entity.Subscription;
+import com.github.bumblebee.command.youtube.service.YoutubeSubscriptionService;
+import com.github.bumblebee.service.RandomPhraseService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import telegram.api.BotApi;
+import telegram.domain.Update;
+
+import java.util.*;
+
+
+@Component
+public class YoutubeSubscribeCommand extends SingleArgumentCommand {
+
+    private final BotApi botApi;
+    private final YoutubeSubscriptionService service;
+    private final RandomPhraseService randomPhraseService;
+    private final YoutubeSubscriptionProvider provider;
+    private final List<Subscription> subscriptionList;
+
+    @Autowired
+    public YoutubeSubscribeCommand(BotApi botApi,
+                                   YoutubeSubscriptionService service,
+                                   RandomPhraseService randomPhraseService,
+                                   YoutubeSubscriptionProvider provider) {
+        this.botApi = botApi;
+        this.service = service;
+        this.randomPhraseService = randomPhraseService;
+        this.provider = provider;
+        this.subscriptionList = service.retrieveSubscriptions();
+    }
+
+    public Set<Long> getChatIds(String channelId) {
+        Set<Long> idSet = new HashSet<>();
+        for (Subscription sub : subscriptionList) {
+            if (sub.getChannelId().equals(channelId)) {
+                for (Chat chat : sub.getChats()) {
+                    idSet.add(chat.getChatId());
+                }
+            }
+        }
+        return idSet;
+    }
+
+    public List<Subscription> getSubscriptionList() {
+        return subscriptionList;
+    }
+
+    @Override
+    public void handleCommand(Update update, Long chatId, String argument) {
```

Comment:
Feel free to rename "argument" parameter to more context related one. For example: "channelId".